python/ray/util/scheduling_strategies.py

```diff
@@ -40,6 +41,16 @@ def __init__(
         self.placement_group_capture_child_tasks = placement_group_capture_child_tasks
 
 
+@PublicAPI(stability="beta")
+class NodeSchedulingStrategy:
+    def __init__(self, node_id: "ray.NodeID", soft: bool):
```

Comment:
I'm not sure we want to expose the NodeID type in a public API. How about standardizing on the hex string? We can also make the same change for runtime_context (expose only hex string).